ingestors/kafka/src/main/scala/hydra/kafka/algebras/KafkaClientAlgebra.scala

```diff
@@ -0,0 +1,187 @@
+  ): F[Either[PublishError, Unit]]
+
+
+  /**
+    * Consume the Hydra record from Kafka.
+    * Does not commit offsets. Each time function is called will return
+    * @param topicName - topic name to consume
+    * @param consumerGroup - group id for consume
+    * @return Stream that results in tupled K and V
+    */
+  def consumeMessages(
+     topicName: TopicName,
+     consumerGroup: ConsumerGroup
+   ): fs2.Stream[F, Record]
+
+}
+
+object KafkaClientAlgebra {
+
+  type TopicName = String
+  type ConsumerGroup = String
+  type Record = (GenericRecord, GenericRecord)
+
+  sealed abstract class PublishError(message: String)
+    extends Exception(message)
+      with Product
+      with Serializable
+
+  object PublishError {
+
+    final case object Timeout
+      extends PublishError("Timeout while ingesting message.")
+        with NoStackTrace
+
+  }
+
+
+  private def getProducerQueue[F[_]: ConcurrentEffect: ContextShift]
+  (bootstrapServers: String, schemaRegistryClient: SchemaRegistryClient): F[fs2.concurrent.Queue[F, (GenericRecord, GenericRecord, TopicName, Deferred[F, Unit])]] = {
+    import fs2.kafka._
+    val producerSettings =
+      ProducerSettings(keySerializer = getGenericRecordSerializer(schemaRegistryClient)(isKey = true), valueSerializer = getGenericRecordSerializer(schemaRegistryClient)())
+        .withBootstrapServers(bootstrapServers)
+    for {
+      queue <- fs2.concurrent.Queue.unbounded[F, (GenericRecord, GenericRecord, TopicName, Deferred[F, Unit])]
+      _ <- Concurrent[F].start(queue.dequeue.map { payload =>
+        val record = ProducerRecord(payload._3, payload._1, payload._2)
+        ProducerRecords.one(record, payload._4)
+      }.through(produce(producerSettings)).evalMap { i => i.passthrough.complete(()) }.compile.drain)
+    } yield queue
+  }
+
+  def live[F[_]: ContextShift: ConcurrentEffect: Timer](
+      bootstrapServers: String,
+      schemaRegistryAlgebra: SchemaRegistry[F]
+  ): F[KafkaClientAlgebra[F]] = schemaRegistryAlgebra.getSchemaRegistryClient.flatMap { schemaRegistryClient =>
+    getProducerQueue[F](bootstrapServers, schemaRegistryClient).map { queue =>
+      new KafkaClientAlgebra[F] {
+        override def publishMessage(
+                                     record: Record,
+                                     topicName: TopicName
+                                   ): F[Either[PublishError, Unit]] = {
+          Deferred[F, Unit].flatMap { d =>
+            queue.enqueue1((record._1, record._2, topicName, d)) *>
+              Concurrent.timeoutTo[F, Either[PublishError, Unit]](d.get.map(Right(_)), 5.seconds, Sync[F].pure(Left(PublishError.Timeout)))
+          }
+        }
+
+        override def consumeMessages(topicName: TopicName, consumerGroup: String): fs2.Stream[F, (GenericRecord, GenericRecord)] = {
+          val consumerSettings = ConsumerSettings(
+            keyDeserializer = getGenericRecordDeserializer(schemaRegistryClient)(isKey = true),
```

Comment:
You could switch this up to be:

```scala
val getDeserializer = getGenericRecordDeserializer(schemaRegistryClient)
val consumerSettings = ConsumerSettings(
            keyDeserializer = getDeserializer(isKey = true),
            valueDeserializer = getDeserializer()
          )
```

Otherwise, you could probably just get rid of the currying.

Not an essential change to make to merge this, just some thoughts.